clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -366,7 +384,7 @@ public void close() {
         }
     }
 
-    public void abort() {
+    public void abortRecordAppends() {
```

Comment:
It may be worth adding a comment explaining how `abort` and `abortRecordAppends` differ in one of the methods and add a reference in the other one. It seems like the reason we have two separate methods is that we do one of them with the lock held and the other with no lock held (as it invokes callbacks).